can/bus.py

```diff
@@ -420,14 +421,27 @@ def shutdown(self) -> None:
         Called to carry out any interface specific cleanup required
         in shutting down a bus.
         """
+        if self._is_shutdown:
+            LOG.debug(f"{self.__class__} is already shut down")
```

Comment:
initially I wanted to raise an exception here but there are existing unit tests that seem to want to preserve the ability to call `shutdown` multiple times without error

Reply:
Yes, I think this needs to be possible. :) Do all subclasses of `BusABC` call `super().shutdown()`? I think this is a good point in time to make sure they do, such that they properly handly multiple calls to `shutdown()`.

Reply:
Added a unittest to check and preserve this behavior.

There were a couple missing such as `socketcand` and `ixxat` which I added commits for.

It's also missing from`etas` but I'm unsure whether it should call `super().shutdown()` as well? I ask because the implementation looks vastly different than the rest.

Reply:
Regarding `etas`: I think it should not be an issue to add it here: https://github.com/hardbyte/python-can/blob/5c523ec9cc5ab3badbb6def6fb3750d228c7c7c0/can/interfaces/etas/__init__.py#L250?

Reply:
The test is a neat idea 😄 So you also checked the code manually? Because due to imports etc. not all all interfaces might run through that test.

Reply:
Thanks! I did verify within `etas` and `gs_usb` which appear to be the only 2 that are "missing coverage" so to speak. 

I will get that added to the `etas` interface :+1: